src/main/java/com/maxi/calendar/controller/EventController.java

```diff
@@ -0,0 +1,153 @@
+package com.maxi.calendar.controller;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.maxi.calendar.model.Event;
+import com.maxi.calendar.model.User;
+import com.maxi.calendar.service.IEventService;
+import com.maxi.calendar.service.IUserService;
+
+@RestController
+@RequestMapping("/events")
+public class EventController {
+	
+	@Autowired
+	IEventService 		eventService;
+	@Autowired
+	IUserService		userService;
+	
+	@RequestMapping(value="/", method=RequestMethod.GET)
+	public ResponseEntity<?> getAll(){
+		
+		List<Event> events = eventService.getAllEvents();
+		if(events.isEmpty()) {
+			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+		}
+		return new ResponseEntity<>(events, HttpStatus.OK);
+	}
+	
+	@RequestMapping(value="/{id}", method=RequestMethod.GET)
+	public ResponseEntity<?> getEvent(@PathVariable int id) {
+		
+		Event event = eventService.getEvent(id);
+		
+		if (event == null) {
+			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+		}
+		
+		return new ResponseEntity<>(event, HttpStatus.OK);
+		
+	}
+	
+	@RequestMapping(value="/add", method=RequestMethod.POST)
+	public ResponseEntity<?> addEvent(@RequestBody Event newEvent) {
+		
+		Event event = eventService.createEvent(newEvent);
+		
+		return new ResponseEntity<>(event, HttpStatus.CREATED);
+	}
+	
+	@RequestMapping(value="/{id}", method=RequestMethod.PUT)
+	public ResponseEntity<?> editEvent(@RequestBody Event uEvent, @PathVariable int id) {
+		
+		Date uDay = uEvent.getDay();
+		Date uBeginTime = uEvent.getBeginTime();
+		Date uEndTime = uEvent.getEndTime();
+		int  uName = uEvent.getStatus();
+		
+		Event event = eventService.getEvent(id);
+		
+		if (event != null) {
```

Comment:
all this logic should be on the update service

Reply:
set logic moved to serviceImpl